### PR TITLE
Feature/send buildparameters

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Add build parameter to jenkins repository configuration which will be send on each repository post receive hook request.

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] PR is well described
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [x] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: src/main/java/sonia/scm/jenkins/JenkinsConfiguration.java [hunk 1]
```diff
@@ -54,8 +54,7 @@
   private String url;
   private String username;
   private boolean csrf;
-
-  //~--- get methods ----------------------------------------------------------
+  private Set<BuildParameter> buildParameters;
```
Comment (Pilopa) [inline src/main/java/sonia/scm/jenkins/JenkinsConfiguration.java:57]: A set is used here, but the referenced class does not have isEqual or hash code methods.

File: src/main/java/sonia/scm/jenkins/JenkinsConfigurationDto.java [hunk 2]
```diff
@@ -39,6 +44,7 @@
   private String url;
   private String username;
   private boolean csrf;
+  private Set<BuildParameterDto> buildParameters;
```
Comment (Pilopa) [inline src/main/java/sonia/scm/jenkins/JenkinsConfigurationDto.java:47]: A set is used here, but the referenced class does not have isEqual or hash code methods.

File: package.json [hunk 3]
```diff
@@ -1,56 +1,58 @@
...
-    "@types/fetch-mock" : "^7.3.1",
-    "@types/i18next" : "^13.0.0",
-    "@types/jest" : "^24.0.19",
-    "@types/query-string" : "5",
-    "@types/react" : "^16.9.9",
-    "@types/react-redux" : "5.0.7",
-    "@types/react-router-dom" : "^5.1.0",
-    "@types/styled-components" : "^4.1.19",
-    "jest" : "^24.9.0"
+  "devDependencies": {
```
Comment (Pilopa) [inline package.json:37]: Do we need the suffix in the @scm-manager dependencies or would a simple "^2.3.0" suffice ?

File: package.json [hunk 4]
```diff
@@ -1,56 +1,58 @@
...
-    "@scm-manager/ui-extensions" : "^2.0.0",
-    "classnames" : "^2.2.6",
-    "query-string" : "^5.0.1",
-    "react" : "^16.10.2",
-    "react-i18next" : "^10.13.1",
-    "react-redux" : "^5.0.7",
-    "react-router-dom" : "^5.1.2",
-    "redux" : "^4.0.0",
-    "styled-components" : "^4.4.0"
+  "dependencies": {
```
Comment (Pilopa) [inline package.json:12]: Do we need the suffix in the @scm-manager dependencies or would a simple "^2.3.0" suffice ?

File: src/main/resources/locales/en/plugins.json [hunk 5]
```diff
@@ -34,7 +34,15 @@
         "branchesHelp": "List of branches.",
         "branches": "New Branch",
         "branchesAdd": "Add branch",
-        "branchesDelete": "Delete"
+        "branchesDelete": "Delete",
+        "parameterLabel": "Build Job Parameter",
+        "parameterHelpText": "The build job parameters will be send on each request.\nJenkins can only process parameter which are defined in the build job.\nParameter names are case sensitive!",
```
Comment (Pilopa) [inline src/main/resources/locales/en/plugins.json:39]: missing s on process paremeters

File: src/main/resources/locales/en/plugins.json [hunk 6]
```diff
@@ -34,7 +34,15 @@
...
+        "parameterName": "Name",
+        "parameterNameHelp": "The name of the parameter.\nIt must match the defined build job parameter name in jenkins exactly.",
+        "parameterValue": "Value",
+        "parameterValueHelp": "The value of the parameter:\n - Use \"true\" oder \"false\" as boolean parameters\n - To use a Run Parameter, the value should be in the format \"jobname#buildNumber\"",
```
Comment (Pilopa) [inline src/main/resources/locales/en/plugins.json:43]: replace German "oder" with English "or"

File: src/test/java/sonia/scm/jenkins/JenkinsRepositoryHookHandlerTest.java [hunk 7]
```diff
@@ -24,71 +24,111 @@
```
Comment (Pilopa) [inline src/test/java/sonia/scm/jenkins/JenkinsRepositoryHookHandlerTest.java:24]: Maybe add a negative Test when rubbish is being send to the server ?

File: package.json (hunk 3 above)
Comment (eheimbuch) [reply package.json:37]: This is auto generated.

File: package.json (hunk 4 above)
Comment (eheimbuch) [reply package.json:12]: This is auto generated.

File: src/test/java/sonia/scm/jenkins/JenkinsRepositoryHookHandlerTest.java (hunk 7 above)
Comment (eheimbuch) [reply src/test/java/sonia/scm/jenkins/JenkinsRepositoryHookHandlerTest.java:24]: How should this test look like? The SCM-Manager cannot check if the parameters are valid.

**Pull request merged**